Fix utils import issue for nightly tests

alisonshao opened the pull request:
## Summary
Fix the `ModuleNotFoundError: No module named 'utils'` error in nightly tests.

Several nightly LoRA tests import from `utils`, but the file is located at `test/srt/lora/utils.py` which is not in the Python path when running from `test/nightly/`.

## Note
The original sgl-kernel version mismatch issue is being addressed by #13951 (bot bump workflow).

## Test plan
- [ ] Verify nightly LoRA tests can import utils module
- [ ] Verify nightly tests pass

alisonshao committed:
Fix sgl-kernel version in python/pyproject.toml for torch 2.9 compatibility

Update to sgl-kernel==0.3.18 which is the latest version available on PyPI
and is compatible with torch 2.9.

The previous version 0.3.17.post2 was incompatible with torch 2.9, causing
the "undefined symbol: _ZNK3c106SymInt6sym_neERKS0_" error in nightly tests.

alisonshao committed:
Add shared utils.py for nightly tests

- Copy lora test utils to test/utils.py for shared access
- Update run_suite_nightly.py to add test/ to PYTHONPATH

This fixes "ModuleNotFoundError: No module named 'utils'" in nightly tests.

Fridge003 commented on python/pyproject.toml:
```diff
@@ -57,7 +57,7 @@ dependencies = [
   "sentencepiece",
   "setproctitle",
-  "sgl-kernel==0.3.17.post2",
+  "sgl-kernel==0.3.18",
```
Should be 0.3.18.post1 here

Fridge003 commented:
The "bump sgl-kernel to sglang" workflow is missing. Shouldn't we use that one?
Since the version in other places (like dockerfile) also needs to be fixed

alisonshao committed:
Build sgl-kernel from source in nightly tests for torch 2.9 compatibility

The sgl-kernel 0.3.18 on PyPI was built before torch 2.9 was merged,
causing ABI incompatibility errors. This adds a build step to compile
sgl-kernel from source against the current torch version.

Changes:
- Add sgl-kernel-build job to build wheel from source
- All test jobs now depend on sgl-kernel-build
- All test jobs download the wheel artifact and use CUSTOM_BUILD_SGL_KERNEL=true

Kangyan-Zhou commented:
@Fridge003 @alisonshao created fixed the github workflow issue and created https://github.com/sgl-project/sglang/pull/13951

alisonshao committed:
Address review comments: bump sgl-kernel to 0.3.18.post1, remove utils.py

- Update sgl-kernel version from 0.3.18 to 0.3.18.post1
- Remove test/utils.py (will address import issue in separate PR)
- Remove PYTHONPATH changes from run_suite_nightly.py

alisonshao committed:
Fix utils import for nightly LoRA tests

- Copy test/srt/lora/utils.py to test/utils.py
- Add test/ to PYTHONPATH in run_suite_nightly.py

This fixes: ModuleNotFoundError: No module named 'utils'

alisonshao committed:
Revert sgl-kernel build from source changes

The sgl-kernel version bump is handled by #13951 (bot bump workflow),
so we no longer need to build from source in the nightly workflow.

alisonshao committed:
Merge branch 'main' into fix/sgl-kernel-version-pyproject

alisonshao committed:
Rename utils.py to lora_utils.py for clarity

- Rename test/utils.py to test/lora_utils.py to avoid conflicts with
  other utils files (e.g., test/srt/cpu/utils.py)
- Update nightly LoRA test imports to use lora_utils
- Compatible with future migration to test/registered

alisonshao committed:
Fix isort formatting for lora_utils import

alisonshao commented:
https://github.com/sgl-project/sglang/actions/runs/19690528758/job/56405080135
The following tests passed with this fix: 
test_lora_qwen3.py
test_lora_radix_cache.py

alisonshao committed:
Remove copyright and license comments

Removed copyright notice and license information from the file.

alisonshao commented on test/lora_utils.py:
```diff
@@ -0,0 +1,573 @@
+# Copyright 2023-2024 SGLang Team
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
```
done

alisonshao committed:
Consolidate LoRA test utils to avoid duplicate code

Update all LoRA tests to use the central test/lora_utils.py instead of
test/srt/lora/utils.py, and delete the duplicate file.

alisonshao committed:
Fix isort formatting

alisonshao committed:
Merge branch 'main' into fix/sgl-kernel-version-pyproject

alisonshao committed:
Add sys.path for lora_utils import in srt/lora tests

alisonshao committed:
Merge branch 'main' into fix/sgl-kernel-version-pyproject

alisonshao reviewed (commented):
Consider adding a TODO comment to indicate this can be cleaned up later:

```python
# Add test directory to path for lora_utils import
# TODO: can be removed after migration
sys.path.insert(0, str(Path(__file__).resolve().parent.parent.parent))
```

This applies to the 4 files: test_lora.py, test_lora_backend.py, test_lora_tp.py, test_multi_lora_backend.py

alisonshao committed:
Add TODO comment for path hack removal after migration

alisonshao committed:
Merge branch 'main' into fix/sgl-kernel-version-pyproject

alisonshao committed:
Add sys.path hack for lora_utils import in nightly tests

alisonshao committed:
Merge branch 'main' into fix/sgl-kernel-version-pyproject
